source/client.c

```diff
@@ -173,6 +173,40 @@ static void s_init_statistics(struct aws_mqtt_connection_operation_statistics_im
     aws_atomic_store_int(&stats->unacked_operation_size_atomic, 0);
 }
 
+static bool s_is_topic_shared_topic(struct aws_byte_cursor *input) {
+    char *input_str = (char *)input->ptr;
+    if (strncmp("$share", input_str, strlen("$share")) == 0) {
+        return true;
+    }
+    return false;
+}
+
+static struct aws_string *s_get_normal_topic_from_shared_topic(struct aws_string *input) {
+    const char *input_char_str = aws_string_c_str(input);
+    int input_char_length = strlen(input_char_str);
+    int found_slashes = 0;
+    int split_position = 0;
```

Comment:
trivial:
When this is called, we're certain that the input starts with "$share" so we could start the check at index 7 and construct the return result at the first encountered '/' or fall from the while directly to the ERROR and return NULL.